internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -234,13 +236,133 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 	}
 
 	return r.updateStatus(ctx, log, clusterInstall, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) setHTTPSIgnitionURL(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster, clusterInstall *hiveext.AgentClusterInstall, secretRef types.NamespacedName) error {
+
+	if clusterInstall.Spec.IgnitionEndpoint == nil {
+		log.Debugf("Empty endpoint found")
+		// Set custom Ignition endpoint so that day2 workers would join using HTTPS endpoint
+		// ensureMCSCert would add ignition override with MCS secret
+		ignitionEndpoint := fmt.Sprintf("%s/config/%s", common.GetMCSUrlBase(cluster), models.HostRoleWorker)
+		log.Infof("setting worker ignition URL to %s", ignitionEndpoint)
+
+		freshClusterInstall := &hiveext.AgentClusterInstall{}
+		key := types.NamespacedName{
+			Namespace: clusterInstall.Namespace,
+			Name:      clusterInstall.Name,
+		}
+		if err := r.Get(ctx, key, freshClusterInstall); err != nil {
+			log.WithError(err).Errorf("Failed to get resource %s", key)
+			return client.IgnoreNotFound(err)
+		}
+
+		freshClusterInstall.Spec.IgnitionEndpoint = &hiveext.IgnitionEndpoint{
+			Url: ignitionEndpoint,
+			CaCertificateReference: &hiveext.CaCertificateReference{
+				Name:      secretRef.Name,
+				Namespace: secretRef.Namespace,
+			},
+		}
+		if err := r.Update(ctx, freshClusterInstall); err != nil {
+			return err
+		}
+		log.Debugf("Endpoint updated")
+	}
+	return nil
+}
+
+func (r *ClusterDeploymentsReconciler) createMCSCertSecret(ctx context.Context, log logrus.FieldLogger, clusterDeployment *hivev1.ClusterDeployment, cluster *common.Cluster, clusterInstall *hiveext.AgentClusterInstall) (types.NamespacedName, error) {
+
+	log.Debugf("Fetching MCS secret from spoke cluster")
+
+	key := types.NamespacedName{
+		Namespace: clusterDeployment.Namespace,
+		Name:      fmt.Sprintf(mcsSecretStringTemplate, clusterDeployment.Name),
+	}
+	log.Debugf("Expected MCS secret key: %s", key)
+	secret, getErr := getSecret(ctx, r.Client, r.APIReader, key)
+	log.Debugf("Fetched MCS secret %v err %v", secret, getErr)
+
+	if getErr != nil && !k8serrors.IsNotFound(getErr) {
+		log.WithError(getErr).Errorf("failed to get secret %s", key)
+		return key, getErr
+	}
+	// Exit early if the secret already exists
+	if getErr == nil {
+		return key, nil
+	}
+
+	kubeConfigSecretKey := types.NamespacedName{
+		Namespace: clusterDeployment.Namespace,
+		Name:      getClusterDeploymentAdminKubeConfigSecretName(clusterDeployment),
+	}
+	log.Debugf("Kubeconfig secret key: %s", kubeConfigSecretKey)
+
+	kubeConfigSecret, getErr := getSecret(ctx, r.Client, r.APIReader, kubeConfigSecretKey)
+	log.Debugf("Fetched kubeconfig secret err %v", getErr)
+
+	if getErr != nil {
+		log.WithError(getErr).Errorf("failed to get secret %s", key)
+		return key, getErr
+	}
+
+	spokeClient, err := r.SpokeK8sClientFactory.Create(kubeConfigSecret)
+	log.Debugf("Created spokeClient %v err %v", spokeClient, err)
```

Comment:
not sure why some tests runs get stuck here with
```
time="2022-07-20T14:47:08Z" level=warning msg="failed to get spoke kube client" func="github.com/openshift/assisted-service/internal/controller/controllers.(*spokeK8sClientFactory).Create" file="/assisted-service/internal/controller/controllers/spoke_k8s_client.go:83" error="Get \"https://api.assisted-test-cluster.redhat.com:6443/api?timeout=32s\": dial tcp: lookup api.assisted-test-cluster.redhat.com on 172.30.0.10:53: no such host"
```
error, and some runs wipe away APIVIP/IngressVIP after `setHTTPSIgnitionURL`